enable background location support in osm build

bemdark opened the pull request:
Fixes #1199

Doesn't request background location from the user (as there is no good UI for it), but if you enable it yourself in the settings, owntracks doesn't have to be started manually when rebooting the phone.

growse commented:
Thanks for this - been meaning to get around to looking at this. 

Hopefully later on we can have the OSS flavour actually request background permission as well, but that'll take a bit more thinking about.

bemdark commented:
Well it's not like it was lot of work, basically I just added the permission to the manifest.
I have tried it though and added the permission via the settings, and it seems to work without changing anything else (I am not really well-versed in how background and foreground services work and how they differ). I added the permission and after restarting the phone I received locations from my phone on another device without ever actually starting OwnTracks manually.

I tried how it looks like to request the background permission - you just get a small warning in the bottom that some permissions are missing and a "Fix" button that sends you to the settings (OwnTracks app details page), so I'm not sure if that's worth implementing at all (I think I'd rather implement it like disabling battery optimization on gms).
